packages/orchestration/README.md

```diff
@@ -131,28 +130,53 @@ const response = await orchestrationClient.chatCompletion({
 });
 
 const responseContent = response.getContent();
+const finishReason = response.getFinishReason();
+const tokenUsage = response.getTokenUsage();
 ```
 
-`getContent()` is a convenience method that parses the response and returns the model's output as a string.
+You can use the following convenience methods for handling chat completion responses:
 
-To retrieve the `finish_reason` for stopping the chat completion request, use the convenience method `getFinishReason()`:
+- `getContent()`: parses the response and returns the model's output as a string.
+- `getFinishReason()`: retrieves the finish_reason, explaining why the chat completion request stopped.
+- `getTokenUsage()`: provides token usage details, namely `total_tokens`, `prompt_tokens`, and `completion_tokens`.
 
-```ts
-const finishReason = response.getFinishReason();
-```
+#### Structured Outputs
 
-Use the `getTokenUsage()` convenience method to retrieve the token usage details of the chat completion request:
+The `response_format` under `templating` guarantees that the model's output aligns with the JSON schemas specified by developers.
+It is useful when the model is not calling a tool, but rather, responding to the user in a structured way.
+By setting `strict: true`, the model ensures that its outputs conform precisely to the provided schema.
 
 ```ts
-const tokenUsage = response.getTokenUsage();
-
-console.log(
-  `Total tokens consumed by the request: ${tokenUsage.total_tokens}\n` +
-    `Input prompt tokens consumed: ${tokenUsage.prompt_tokens}\n` +
-    `Output text completion tokens consumed: ${tokenUsage.completion_tokens}\n`
-);
+templating: {
+    template: [
+      { role: 'user', content: 'What is the capital of {{?country}}?' }
+    ],
+    response_format: {
+        type: 'json_schema',
+        json_schema: {
+          name: 'capital_response',
+          strict: true,
+          schema: {
+            type: 'object',
+            properties: {
+              country_name: {
+                type: "string",
+                description: "The name of the country provided by the user."
+              },
+              capital: {
+                type: "string",
+                description: "The capital city of the country."
+              }
+            },
+            required: ["country_name", "capital"]
+          }
+        }
+      }
+  }
 ```
 
+In addition to using JSON schema objects, a Zod schema is also supported.
```

Comment:
[req] Add a link to sample code? Or provide an example here.

Reply:
I was actually debating this as well. I am not a fan of linking code sample lines since they are often bound to change. And adding one more detailed example felt a bit too detailed. So I decided to add this note, the user can then google it themselves (I could add a note admonition to highlight it)
Do you have a strong preference?

Reply:
For me personally, it was a bit unclear where to put zod. But after having a glimpse at the sample code, I immediately understood that zod can construct such jaon schema object instead of manually writing the schema. Maybe we can provide a hint of using zodToJsonSchema() function and where to put it. I don't prefer linking sample code here, I only wanna point out that a small step is missing here.